webapp/src/lib/dialog-transaction.ts

```diff
@@ -14,72 +23,59 @@ export interface IDialogTransactionConfig<T> {
     styleUrls: ["./dialog-transaction.scss"],
     templateUrl: "dialog-transaction.html",
 })
-export class DialogTransactionComponent<T> implements OnInit {
+export class DialogTransactionComponent<T> implements AfterViewInit {
 
     percentage: number = 0;
-    blockIndex: number;
     text: string;
     error: Error | undefined;
 
     private blocks: Element[] = [];
     private transaction: Element;
+    private ub: Subscription;
     @ViewChild("main", {static: false}) private main?: ElementRef;
-    private ubTimer: Timer;
 
     constructor(
+        private bcs: ByzCoinService,
         readonly dialogRef: MatDialogRef<DialogTransactionComponent<T>>,
-        private readonly bcv: BcviewerService,
         private readonly renderer: Renderer2,
         @Inject(MAT_DIALOG_DATA) public data: IDialogTransactionConfig<T>) {
     }
 
-    async ngOnInit() {
-        this.updateBlocks();
-        if (!this.bcv.currentBlock) {
-            await new Promise((resolve) => {
-                this.bcv.newStatus.subscribe(() => {
-                    resolve();
-                });
-            });
-        }
-        this.ubTimer = setInterval(() => this.updateBlocks(), 500);
-        setTimeout(() => this.startTransactions(), 200);
+    async ngAfterViewInit() {
+        // TODO: replace the setTimeout with ChangeDetectorRef
+        setTimeout(async () => {
```

Comment:
why using a `setTimeout` and not awaiting or launching it directly?

Reply:
there was an error when creating the dialog that had to do with changing the layout while it's still drawing it. But I cannot reproduce it anymore. So:
- `ngAfterViewInit` -> `ngOnInit`
- removed `setTimeout`